### PR TITLE
python311Packages.jaxtyping: 0.2.25 -> 0.2.26

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Update `jaxtyping`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 291421` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.ax</li>
    <li>python311Packages.ax.dist</li>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.gpytorch</li>
    <li>python311Packages.gpytorch.dist</li>
    <li>python311Packages.jaxtyping</li>
    <li>python311Packages.jaxtyping.dist</li>
    <li>python311Packages.linear-operator</li>
    <li>python311Packages.linear-operator.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pyannote-pipeline</li>
    <li>python311Packages.pyannote-pipeline.dist</li>
    <li>python312Packages.jaxtyping</li>
    <li>python312Packages.jaxtyping.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 291421` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pyannote-pipeline</li>
    <li>python311Packages.pyannote-pipeline.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python311Packages.ax</li>
    <li>python311Packages.ax.dist</li>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.gpytorch</li>
    <li>python311Packages.gpytorch.dist</li>
    <li>python311Packages.jaxtyping</li>
    <li>python311Packages.jaxtyping.dist</li>
    <li>python311Packages.linear-operator</li>
    <li>python311Packages.linear-operator.dist</li>
    <li>python312Packages.jaxtyping</li>
    <li>python312Packages.jaxtyping.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/jaxtyping/default.nix [hunk 1]
```diff
@@ -28,12 +28,12 @@ let
...
       repo = "jaxtyping";
       rev = "refs/tags/v${version}";
-      hash = "sha256-+JqpI5xrM7o73LG6oMix88Jr5aptmWYjJQcqUNo7icg=";
+      hash = "sha256-2QDTRNH2/9FPU5xrQx7yZRHwEWqj0PUNzcCuKwY4yNg=";
     };
 
     postPatch = ''
       substituteInPlace pyproject.toml \
-        --replace "typeguard>=2.13.3,<3" "typeguard"
+        --replace "typeguard==2.13.3" "typeguard"
```
Comment (samuela) [inline pkgs/development/python-modules/jaxtyping/default.nix:36]: could `pythonRelaxDepsHook` do this?

Comment (GaetanLepage) [reply pkgs/development/python-modules/jaxtyping/default.nix]: Indeed it works, done !

**Review by samuela — APPROVED** [review]
Diff lgtm

**Pull request merged**